CCL.Importer/Mapper.cs

```diff
@@ -108,47 +144,16 @@ private static void Configure(IMapperConfigurationExpression cfg)
             destination = MapComponent<TSource, TDestination>(source);
         }
 
-        public static void MapMultipleComponents<TSource, TDestination>(IEnumerable<TSource> source, out IEnumerable<TDestination> destination)
-            where TSource : MonoBehaviour
-            where TDestination : MonoBehaviour
-        {
-            List<(TSource Source, TDestination Destination)> destinationList = new();
-
-            // Start by adding all components.
-            foreach (var item in source)
-            {
-                destinationList.Add((item, item.gameObject.AddComponent<TDestination>()));
-            }
-
-            // Once all components are added, it's safe to map them.
-            foreach (var item in destinationList)
-            {
-                M.Map(item.Source, item.Destination);
-                Object.Destroy(item.Source);
-            }
-
-            // Return only the new components.
-            destination = destinationList.Select(x => x.Destination);
-        }
-
-        private static Component GetMappedComponent<TSource>(TSource source)
-            where TSource : MonoBehaviour
+        public static void ClearCache()
         {
-            // https://www.youtube.com/watch?v=rmQFcVR6vEs
-            return source.gameObject.GetComponent(M.ConfigurationProvider.GetAllTypeMaps()
-                    .First(map => map.SourceType == source.GetType()).DestinationType);
-
-            //if (s_componentMapCache.TryGetValue(source, out MonoBehaviour mapped))
-            //{
-            //    return mapped;
-            //}
-
-            //return source;
+            s_componentMapCache.Clear();
         }
 
-        public static void ClearCache()
+        internal static MonoBehaviour GetFromCache(MonoBehaviour source)
```

Comment:
Missing a version of the cache that will return the source instead of null when not found (use case: `PlayerDistanceMultipleGameObjectsOptimizer` might reference both proxies and direct scripts).

Reply:
the more I think about it, the more I don't think such a version should exist.  A localized wrapper maybe, but a function whose return is sometimes what's in the map and sometimes what was passed in opens a large door for bugs down the road.

If you can identify places that need that in the current branch I can update those places to handle the null coming back and adjust as appropriate.

We should identify if this is a common need or is scoped to some very specific scripts - if the former then we'll need more robust handling, if the latter, then those scripts probably should be excluded from the normal mapping process

Reply:
That takes MonoBehaviours directly, I've only seen that one in the example. There's a few that take some less generic MonoBehaviours (`SimConnectionDefinition` with `executionOrder`, `SimController` with `otherSimControllers`).

Reply:
SimConnectionDefinition and SimController we're building (at least at the moment) - so not subject to this conversation.

I think the correct answer here is to fix the few places currently broken and leave it in that state until a more concrete example arrives.